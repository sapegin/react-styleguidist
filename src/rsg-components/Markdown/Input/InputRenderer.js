import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'rsg-components/Styled';

const styles = () => ({
	input: {
		isolate: false,
		display: 'inline-block',
		verticalAlign: 'middle',
	},
});

export function InputRenderer({ classes, ...rest }) {
	return <input {...rest} className={classes.input} />;
}
InputRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
};

export default Styled(styles)(InputRenderer);
