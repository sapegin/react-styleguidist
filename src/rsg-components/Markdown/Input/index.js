export { default } from 'rsg-components/Markdown/Input/InputRenderer';
