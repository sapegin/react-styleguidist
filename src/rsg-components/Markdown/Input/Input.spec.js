import React from 'react';

import Input from './index';

describe('Markdown Input', () => {
	it('should render an input with the specified type', () => {
		const actual = render(<Input type="checkbox" />);

		expect(actual).toMatchSnapshot();
	});
});
