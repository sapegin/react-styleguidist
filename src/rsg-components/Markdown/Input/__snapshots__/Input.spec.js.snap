// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Input should render an input with the specified type 1`] = `
<input
  class="rsg--input-0"
  type="checkbox"
/>
`;
